### PR TITLE
fix attr bug in op_test and ensure order in duplicate inputs/outputs

### LINKED ISSUE #4019: ensure the order of duplicate inputs/outputs
The dict in python is unordered. And in some circumstances, we have two or more duplicate inputs, and the order in duplicate inputs must keep the same. So, we need to change dict to list of tuple to ensure right order.

### PR DESCRIPTION
**Pull request by QiJune**
Fix #4019

### CODE REVIEW

**Comment by Yancey1989** [comment]
Hi @Xreki @QiJune , the CI build failed, maybe we need to merge #4018 first, and pull the develop branch in this PR.

File: python/paddle/v2/framework/tests/op_test.py [hunk 1]
```diff
@@ -9,15 +9,15 @@ def grad_var_name(var_name):
...
+def create_op(scope, op_type, inputs, outputs, attrs):
     kwargs = dict()
 
     for in_name, in_dup in Operator.get_op_inputs(op_type):
         if in_name in inputs:
             kwargs[in_name] = []
             if in_dup:
                 sub_in = inputs[in_name]
-                for sub_in_name in sub_in:
+                for sub_in_name, arr in sub_in:
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/op_test.py]:
```python
 for sub_in_name, _ in sub_in:
```

File: python/paddle/v2/framework/tests/op_test.py [hunk 2]
```diff
@@ -29,15 +29,16 @@ def create_op(scope, op_type, inputs, outputs, attrs=None):
             kwargs[out_name] = []
             if out_dup:
                 sub_in = outputs[out_name]
-                for sun_in_name in sub_in:
-                    var = scope.new_var(sun_in_name)
-                    kwargs[out_name].append(sun_in_name)
+                for sub_in_name, arr in sub_in:
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/op_test.py]:
```python
for sub_in_name, _ in sub_in:
```

File: python/paddle/v2/framework/tests/op_test.py [hunk 3]
```diff
@@ -46,10 +47,9 @@ def set_input(scope, op, inputs, place):
         if in_name in inputs:
             if in_dup:
                 sub_in = inputs[in_name]
-                for sub_in_name in sub_in:
+                for sub_in_name, arr in sub_in:
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/op_test.py]: Make name more clear.

```
 for sub_in_name, sub_array in sub_in:
```

File: python/paddle/v2/framework/tests/op_test.py [hunk 4]
```diff
@@ -168,7 +168,11 @@ def get_gradient(scope, op, inputs, outputs, grad_name, place,
 class OpTest(unittest.TestCase):
     def check_output_with_place(self, place):
         self.scope = core.Scope()
-        self.op = create_op(self.scope, self.op_type, self.inputs, self.outputs)
+        op_inputs = self.inputs if hasattr(self, "inputs") else dict()
+        op_outputs = self.outputs if hasattr(self, "outputs") else dict()
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/op_test.py]: When the users could not set `self.inputs`  and `self.outputs` ?  Aren't they always needed?

**Comment by typhoonzero** [comment]
Need to add "Fix [issue url]" in the PR description.

File: python/paddle/v2/framework/tests/op_test.py (hunk 4 above)
Comment (QiJune) [reply python/paddle/v2/framework/tests/op_test.py]: Some operator, like random operator has no input. NOP has no input and output. 
I will remove self.outputs here.

File: python/paddle/v2/framework/tests/op_test.py (hunk 2 above)
Comment (QiJune) [reply python/paddle/v2/framework/tests/op_test.py]: Done

File: python/paddle/v2/framework/tests/op_test.py (hunk 3 above)
Comment (QiJune) [reply python/paddle/v2/framework/tests/op_test.py]: Done

File: python/paddle/v2/framework/tests/op_test.py (hunk 1 above)
Comment (QiJune) [reply python/paddle/v2/framework/tests/op_test.py]: Done

**Pull request merged**